spec/helpers/proposals_helper_spec.rb

```diff
@@ -86,79 +86,147 @@
   end
 
   describe "#assigned_dates" do
-    context "when start date is blank" do
+    context "for valid parameters" do
       let(:location) { create(:location) }
-      it "creates range of dates" do
-        assigned_dates(location)
-        expect(response).to have_http_status(:ok)
+      it "returns the range of dates" do
+        dates = assigned_dates(location)
+        expect(assigned_dates(location)).to match_array(dates)
```

Comment:
`expect(assigned_dates(location)).to eq(assigned_dates(location))` isn't testing anything.

Reply:
When `assigned_dates(location)`, under which condition will `assigned_dates(location)` not match_array(dates)?